bonsai-core/src/main/scala/com/stripe/bonsai/Layout.scala

```diff
@@ -4,31 +4,45 @@ import scala.language.implicitConversions
 
 import scala.reflect.ClassTag
 
+import java.io.{ DataOutput, DataInput }
+
 import com.stripe.bonsai.layout._
 
 trait Layout[A] {
   def newBuilder: VecBuilder[A]
 
+  def write(vec: Vec[A], out: DataOutput): Unit
+
+  def read(in: DataInput): Vec[A]
+
+  def isSafeToCast(vec: Vec[_]): Boolean
+
```

Comment:
OK, I guess let's leave it for now and we'll see where we end up. It's not a huge amount of duplication.